### PR TITLE
v4 fluent, support x-ms-original-file in latest m4

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
- core and m4 version update
- rename `xmsOriginalFile` method to `originalFile`
- generate sample with `x-ms-original-file` metadata

Sample:
```java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Code generated by Microsoft (R) AutoRest Code Generator.

package com.azure.mgmtlitetest.storage.generated;

import com.azure.mgmtlitetest.storage.models.CorsRule;
import com.azure.mgmtlitetest.storage.models.CorsRuleAllowedMethodsItem;
import com.azure.mgmtlitetest.storage.models.CorsRules;
import com.azure.mgmtlitetest.storage.models.Multichannel;
import com.azure.mgmtlitetest.storage.models.ProtocolSettings;
import com.azure.mgmtlitetest.storage.models.SmbSetting;
import java.util.Arrays;

/** Samples for FileServices SetServiceProperties. */
public final class FileServicesSetServicePropertiesSamples {
    /*
     * x-ms-original-file: specification/storage/resource-manager/Microsoft.Storage/stable/2021-01-01/examples/FileServicesPut.json
     */
    /**
     * Sample code: PutFileServices.
     *
     * @param manager Entry point to StorageManager.
     */
    public static void putFileServices(com.azure.mgmtlitetest.storage.StorageManager manager) {
        manager
            .fileServices()
            .define()
            .withExistingStorageAccount("res4410", "sto8607")
            .withCors(
                new CorsRules()
                    .withCorsRules(
                        Arrays
                            .asList(
                                new CorsRule()
                                    .withAllowedOrigins(
                                        Arrays.asList("http://www.contoso.com", "http://www.fabrikam.com"))
                                    .withAllowedMethods(
                                        Arrays
                                            .asList(
                                                CorsRuleAllowedMethodsItem.GET,
                                                CorsRuleAllowedMethodsItem.HEAD,
                                                CorsRuleAllowedMethodsItem.POST,
                                                CorsRuleAllowedMethodsItem.OPTIONS,
                                                CorsRuleAllowedMethodsItem.MERGE,
                                                CorsRuleAllowedMethodsItem.PUT))
                                    .withMaxAgeInSeconds(100)
                                    .withExposedHeaders(Arrays.asList("x-ms-meta-*"))
                                    .withAllowedHeaders(
                                        Arrays.asList("x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*")),
                                new CorsRule()
                                    .withAllowedOrigins(Arrays.asList("*"))
                                    .withAllowedMethods(Arrays.asList(CorsRuleAllowedMethodsItem.GET))
                                    .withMaxAgeInSeconds(2)
                                    .withExposedHeaders(Arrays.asList("*"))
                                    .withAllowedHeaders(Arrays.asList("*")),
                                new CorsRule()
                                    .withAllowedOrigins(
                                        Arrays.asList("http://www.abc23.com", "https://www.fabrikam.com/*"))
                                    .withAllowedMethods(
                                        Arrays.asList(CorsRuleAllowedMethodsItem.GET, CorsRuleAllowedMethodsItem.PUT))
                                    .withMaxAgeInSeconds(2000)
                                    .withExposedHeaders(
                                        Arrays.asList("x-ms-meta-abc", "x-ms-meta-data*", "x-ms-meta-target*"))
                                    .withAllowedHeaders(Arrays.asList("x-ms-meta-12345675754564*")))))
            .create();
    }

    /*
     * x-ms-original-file: specification/storage/resource-manager/Microsoft.Storage/stable/2021-01-01/examples/FileServicesPut_EnableSecureSmbFeatures.json
     */
    /**
     * Sample code: PutFileServices_EnableSecureSmbFeatures.
     *
     * @param manager Entry point to StorageManager.
     */
    public static void putFileServicesEnableSecureSmbFeatures(com.azure.mgmtlitetest.storage.StorageManager manager) {
        manager
            .fileServices()
            .define()
            .withExistingStorageAccount("res4410", "sto8607")
            .withProtocolSettings(
                new ProtocolSettings()
                    .withSmb(
                        new SmbSetting()
                            .withVersions("SMB2.1;SMB3.0;SMB3.1.1")
                            .withAuthenticationMethods("NTLMv2;Kerberos")
                            .withKerberosTicketEncryption("RC4-HMAC;AES-256")
                            .withChannelEncryption("AES-128-CCM;AES-128-GCM;AES-256-GCM")))
            .create();
    }

    /*
     * x-ms-original-file: specification/storage/resource-manager/Microsoft.Storage/stable/2021-01-01/examples/FileServicesPut_EnableSMBMultichannel.json
     */
    /**
     * Sample code: PutFileServices_EnableSMBMultichannel.
     *
     * @param manager Entry point to StorageManager.
     */
    public static void putFileServicesEnableSMBMultichannel(com.azure.mgmtlitetest.storage.StorageManager manager) {
        manager
            .fileServices()
            .define()
            .withExistingStorageAccount("res4410", "sto8607")
            .withProtocolSettings(
                new ProtocolSettings().withSmb(new SmbSetting().withMultichannel(new Multichannel().withEnabled(true))))
            .create();
    }
}
```

Automation will collect on 
```
x-ms-original-file: specification/storage/resource-manager/Microsoft.Storage/stable/2021-01-01/examples/FileServicesPut.json
```

### CODE REVIEW

File: fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java [hunk 1]
```diff
@@ -162,25 +162,23 @@ private FluentExample getExample(Map<String, FluentExample> examples,
...
-                .findFirst()
-                .orElse(null);
-        if (apiVersion == null) {
-            logger.warn("Failed to find api-version in method '{}'", clientMethod.getName());
-        }
-        return apiVersion;
-    }
+//    private static String getApiVersion(ClientMethod clientMethod) {
+//        String apiVersion = clientMethod.getProxyMethod().getParameters().stream()
+//                .filter(p -> "api-version".equals(p.getRequestParameterName()))
```
Comment (haolingdong-msft) [inline fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java:173]: why do we comment this? delete it ?

File: fluentgen/src/test/java/com/azure/autorest/fluent/model/clientmodel/ProxyMethodExampleTests.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Assertions.assertEquals("specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/examples/PutDeploymentAtScope.json", example.getRelativeOriginalFileName());
+
+        example = new ProxyMethodExample.Builder()
+                .originalFile("https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-09-01/examples/getDataPolicyManifest.json")
+                .build();
+        Assertions.assertEquals("specification/resources/resource-manager/Microsoft.Authorization/stable/2020-09-01/examples/getDataPolicyManifest.json", example.getRelativeOriginalFileName());
+
+        // not able to parse
+        example = new ProxyMethodExample.Builder()
+                .originalFile("/c:/github/azure-rest-api-specs/specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/examples/PutDeploymentAtScope.json")
```
Comment (haolingdong-msft) [inline fluentgen/src/test/java/com/azure/autorest/fluent/model/clientmodel/ProxyMethodExampleTests.java:35]: is this local parse? will the test pass in CI?

File: fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java:173]: I am not sure if I might need to use that again. So for now keep it as commented.

File: fluentgen/src/test/java/com/azure/autorest/fluent/model/clientmodel/ProxyMethodExampleTests.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply fluentgen/src/test/java/com/azure/autorest/fluent/model/clientmodel/ProxyMethodExampleTests.java:35]: Local parse. Test pass.

File: fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ExampleParser.java:173]: OK deleted.

File: javagen/src/main/java/com/azure/autorest/model/clientmodel/ProxyMethodExample.java [hunk 3]
```diff
@@ -44,20 +60,85 @@ public String toString() {
...
+                        break;
+                    }
+
+                    default:
+                    {
+//                        LOGGER.error("Unknown protocol in x-ms-original-file: '{}'", originalFileName);
+                    }
+                }
+            } catch (MalformedURLException | URISyntaxException | IllegalArgumentException e) {
+//                LOGGER.error("Failed to parse x-ms-original-file: '{}'", originalFileName);
```
Comment (haolingdong-msft) [inline javagen/src/main/java/com/azure/autorest/model/clientmodel/ProxyMethodExample.java:131]: please check the way to do logging.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**